docs/security/index.md

```diff
@@ -0,0 +1,453 @@
+# Security Plugins
+
+The `SecurityPlugin` interface provides an abstraction around the Telicent Core Platform's label based security model
+allowing applications to enforce the model without needing any direct knowledge of how the labels are interpreted and
+enforced.
+
+## Design
+
+In this section we first outline the design motivation behind the interfaces that are defined, see
+[Interfaces](#interfaces) for outlines of the various interfaces and their usages.  As with everything else in our Core
+Libraries this follows our general [Design Ethos](../design.md#design-ethos).
+
+### Label Format & Syntax
+
+As background, remember that all events flowing through Kafka within the Core Platform are expected to have a
+`Security-Label` header present that indicates the labels that apply to the data.  Additionally, RDF format events on
+the `knowledge` topic *MAY* optionally include a labels graph that provides fine grained labels for individual triple(s)
+within the event.  Currently these labels are expressed as [RDF-ABAC Attribute Expressions][RdfAbac] which are strings
+declaring an Attribute Based Access Control (ABAC) policy in the form of attribute expressions.
+
+As part of evolving the platform towards a more flexible Policy Based Access Control (PBAC) model the new plugin APIs
+introduced from `0.25.0` onwards instead treat the `Security-Label` as opaque byte sequences.  These byte sequences have
+an optional 4 byte schema prefix to allow plugins to detect whether they support a particular label schema and reject
+labels they don't support.  This also potentially provides the capability for future plugins to provide support for
+multiple label schemas coexisting within the platform.
```

Comment:
Why? Managing a single ABAC scheme consistently at an enterprise level is hard enough especially so for multiple co-operating organisations.

Reply:
It's more a theoretical capability for future proofing than anything else.  I don't anticipate us actually supporting this anytime soon.

Also having an explicit schema prefix makes it easier to evolve labelling schemas that don't naturally have their own versioning mechanism built-in e.g. current usage of string based labels.

Reply:
Added clarifications that while this is theoretically supported we have **NO** intention of doing this anytime soon